.github/workflows/test-template-changes.yml

```diff
@@ -0,0 +1,51 @@
+######################################
+## Custom Web Almanac GitHub action ##
+######################################
+#
+# This Action generates the dynamic templates. It also then takes a copy
+# of the templates folder, and then checks out main branch and regenerates the
+# templates and compares the two. If run from a pull request it adds a comment
+# showing the diffs.
+#
+# Note it does not run a full test of the website as that will be completed by
+# the test_website.yml GitHub Action.
+#
+# This Action is useful to check impact of dependency upgrades (e.g. if
+# prettier version change results in different HTML being generated), since we
+# do not store generated templates in Git and so cannot otherwise see these
+# changes. Changes can then be accepted or investigated further before merging.
+#
+name: Test Template Changes
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - '**/package.json'
+      - '**/package-lock.json'
```

Comment:
Is globbing pattern necessary here? If were are watching two specific files and not something nested deep, we might as well specify their paths.

Reply:
No real difference at the moment to be honest. No real harm to glob as only looks at track files (so won’t include node_modules dependencies) and who knows if someone will a separate package.json for a particular subfolder/module in future.